Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -36,14 +36,155 @@ namespace AZ
                 if (loadResult == Data::AssetHandler::LoadResult::LoadComplete)
                 {
                     // ImageMipChainAsset has some internal variables need to initialized after it was loaded.
-                    StreamingImageAsset* assetData2 = asset.GetAs<StreamingImageAsset>();
-                    assetData2->m_tailMipChain.Init();
+                    assetData->m_tailMipChain.Init();
+
+                    // Handle StreamingImageAsset reload (which the asset can be found from asset manager)
+                    auto& assetManager = AZ::Data::AssetManager::Instance();
+                    Data::Asset<StreamingImageAsset> foundImageAsset = assetManager.FindAsset(asset.GetId(), AZ::Data::AssetLoadBehavior::Default);
+                    // If the asset has full mipchain assets, then we want to reload all of them when reload StreamingImageAsset
+                    if (foundImageAsset && foundImageAsset.GetData() != asset.GetData() && foundImageAsset->HasFullMipChainAssets())
```

Comment:
Even if the asset only has _some_ mips, shouldn't those mips still get reloaded? IMO it seems like the "HasFullMipChainAssets" check should only affect whether or not it tries to reload a mip chain asset that isn't currently loaded, not whether or not it reloads at _all_...

Reply:
The reason that I didn't handle some mipchain reload because when an image changed especially it get resized, the mipchain assets number would change. User may not get what they want. For example, if a texture's resolution doubled, the mipchain name and asset id of the mip level 0 could be totally different than the new one. 

We still need to figure out a better way for the user who only want to load one mipchain for their select mip. Before that, user has to load full mipchains to get reliable result.

Reply:
Hm, that's a fair point, though at least in my _opinion_ users are more likely to want the changed image data than the preserved image size. For example, if the Editor has mip 3 of an image loaded at the moment, and the user edits the image from red to blue but twice the resolution, I'd think they would want to see the blue mip even if mip 3 isn't necessarily the right one any more.

But as you say, this can also be revisited later, it's starting to get into a pretty specific nitpicky use case.